### DISCUSSION TITLE
Wrong endpoint exposed for Schneider Electric 545D6102?

### DISCUSSION THREAD

**Discussion by kenni** [Koenkk/zigbee2mqtt#9612] [Devices]
Hi,

I've purchased a Scheider Electric 545D6102, which should be supported according to the device list:
https://www.zigbee2mqtt.io/devices/545D6102.html#schneider%2520electric-545d6102

However, I'm unable to get the current on/off status of the relay, as pressing any of the physical buttons changes the value of "state", while it is my understanding that the zigbee2mqtt device only exposes "state_l1".

The following shows the status when the light is off:
{
    "action": null,
    "ballast_maximum_level_l1": null,
    "ballast_minimum_level_l1": null,
    "ballast_status_lamp_failure": false,
    "ballast_status_non_operational": false,
    "brightness": 1,
    "dimmer_mode": "RC",
    "dimmer_mode_l1": null,
    "level_config": {
        "current_level_startup": "minimum",
        "on_off_transition_time": 0
    },
    "level_config_l1": {
        "current_level_startup": "minimum",
        "on_off_transition_time": 0
    },
    "linkquality": 51,
    **_"state": "OFF",
    "state_l1": "OFF"_**
}

And the following shows the status when I turn on the light by pressing the contact:
{
    "action": null,
    "ballast_maximum_level_l1": null,
    "ballast_minimum_level_l1": null,
    "ballast_status_lamp_failure": false,
    "ballast_status_non_operational": false,
    "brightness": 254,
    "dimmer_mode": "RC",
    "dimmer_mode_l1": null,
    "level_config": {
        "current_level_startup": "minimum",
        "on_off_transition_time": 0
    },
    "level_config_l1": {
        "current_level_startup": "minimum",
        "on_off_transition_time": 0
    },
    "linkquality": 21,
    **_"state": "ON",
    "state_l1": "OFF"_**
}

...and as "state_l1" isn't updated, I'm unable to see that the light has been turned on by someone pressing on the contact.

Toggling the state_l1 from Zigbee2MQTT results in an update of **both** "state" and "state_l1" and correctly turns the light on and off.

Any idea on how to fix this? I suppose that the device needs updating in Zigbee2MQTT? Thanks!

**Comment by CosminLazar** [comment]
This is probably not the answer you are looking for, but I stumbled upon the same issue. After spending some late nights trying to understand how zigbee and zigbee2mqtt work I gave up :)

What I ended up doing is creating an extra entity in home assistant in the configuration file, where I re-use the topics already used by the  Schneider Electric 545D6102 integration as follows:

`light:
  - platform: mqtt
    name: "Kitchen light"
    schema: json
    unique_id: "0x60a423fffe501958_light_real_zigbee2mqtt"
    state_topic: "zigbee2mqtt/kitchen_lights"
    command_topic: "zigbee2mqtt/kitchen_lights/l1/set"
    json_attributes_topic: "zigbee2mqtt/kitchen_lights/l1"
    brightness: true
    brightness_scale: 254
    availability:
      - topic: "zigbee2mqtt/bridge/state"
`

What I am doing here is basically using the l1 topic (zigbee2mqtt/kitchen_lights/l1/set) when I want to control the lights from HomeAssistant and the general switch state when reading the state (zigbee2mqtt/kitchen_lights). 

This works just fine and I can control the device from both HomeAssistant and the physical controls and the state gets updated as expected.

Because I spent some time playing around with this device here are other issues I encountered and tips:

- controlling the brightness by keeping the arrow up and arrow down buttons was very jerky and slow. This was probably because the device was sending zigbee messages for each increment of brightness (aka 254 messages when switching from OFF to ON). I fixed that by setting the Min rep change to 10 (available in the reporting tab of this device in the zigbee2mqtt UI)
- to further fine-tune the brightness use the ballast_minimum_level and ballast_maximum_level
- the bottom row of buttons labeled 1 and 2 are pretty useless by default, as they turn the switch on or off (same behaviour as just pressing th arrow up or down). In order to re-program those buttons I went into the Scene tab and deleted all scenes. Now nothing happens when button 1 or 2 is pressed, but I can capture their press using the exposed action sensor. Now when I press button 1 I trigger a vacuum clean of the kitchen (via nodered).
- if you want the switch to turn on at the previous brightness level, instead of full brightness, then you need to go to Dev Console and write the value `255` to endpoint `3`, cluster `LevelCtrl`, attribute `onLevel`.

**Comment by kenni** [comment]
Hi @CosminLazar,

Thanks for your input. FYI, the missing state change was fixed in https://github.com/Koenkk/zigbee-herdsman-converters/commit/c9473b173bef3f12f1837ce8cffbb49f413f426b

Best regards
Kenni
